functions.cmake

```diff
@@ -228,13 +233,31 @@ function(_add_library)
       add_library(${ARGV} ${VERSION_FILE_PATH})
     else()
       add_library(${ARGV})
+      list(GET ARGV 0 CLR_CMAKE_CURRENT_TARGET)
+      set_target_properties(
+        ${CLR_CMAKE_CURRENT_TARGET} 
+        PROPERTIES
+        VS_GLOBAL_CL_MPCount ${CLR_CMAKE_CPU_CORES})
     endif(NOT WIN32)
     list(FIND CLR_CROSS_COMPONENTS_LIST ${ARGV0} INDEX)  
     if (DEFINED CLR_CROSS_COMPONENTS_LIST AND ${INDEX} EQUAL -1)  
      set_target_properties(${ARGV0} PROPERTIES EXCLUDE_FROM_ALL 1)  
     endif()  
 endfunction()
 
+function(_add_custom_target)
+    if(NOT WIN32)
+      add_custom_target(${ARGV})
+    else()
+      add_custom_target(${ARGV})
+      list(GET ARGV 0 CLR_CMAKE_CURRENT_TARGET)
```

Comment:
Here too, this extraction is not needed, ARGV0 variable can be used to access that argument. See the usage in the code below.

Reply:
Will fix